TanoaPeakSimulator/TanoaPeakSimulator/Tanoa.py

```diff
@@ -0,0 +1,92 @@
+import secrets
+import random
+
+
+def main():
+    location_length = 1000
+    bp_length = 100
+    bp_amount = 200
+
+    genome_file = open("dmel-all-chromosome-r6.38.fasta", "r")
+    sequences = genome_file.read()
+    sequences_arr = sequences.split('>')
+    sequence_object_array = []
+    for sequence in sequences_arr:
+        sequence_split = sequence.splitlines()
+        if len(sequence_split) != 0:
+            header = sequence_split[0]
+            body = ""
+            for x in range(1, len(sequence_split)):
+                body += sequence_split[x]
+            sequence_object_array.append(Chromosome(header, body))
+
+    # for chromosome in sequence_object_array:
```

Comment:
You shouldn't commit commented-out code

Reply:
That's the code to do it to all of the chromosomes. It gives me a memory error when I try to do it that way. That'd be the ideal way to run it though

Reply:
I think you can improve memory efficiency in a few ways:

1. Loop over the lines in the file instead of reading it all at once.  This will allow you to read an entire chromosome, sample the enhancer regions and fake reads, and write those out before you move onto the next chromosome.

2. Look into Python generators.  I often structure code so that say the read function is implemented as a generator, the sampling functions are implemented as generators, and the write function can take a generator.  This way you can still break up the code into reasonably sized functions but everything can be "lazy" so that not everything is loaded into memory at once.